torch/csrc/autograd/engine.cpp

```diff
@@ -729,15 +649,30 @@ auto Engine::execute(const edge_list& roots,
   ClearCallbacks _cb_guard(final_callbacks_, post_callbacks_lock_);
 
   GraphTask graph_task(keep_graph, create_graph, worker_device == NO_DEVICE ? 0 : total_depth+1);
-  // Lock mutex while GraphTask is being set up
-  std::unique_lock<std::mutex> lock(graph_task.mutex_);
 
   // Now compute the dependencies for all executable functions and queue the root
   auto graph_root = std::make_shared<GraphRoot>(roots, inputs);
   compute_dependencies(graph_root.get(), graph_task);
```

Comment:
Can you justify why this doesn't need to operate with the lock? It sure looks like it needs the lock to me...

Reply:
OK, I think I agree that no one can access until the ready queue push.